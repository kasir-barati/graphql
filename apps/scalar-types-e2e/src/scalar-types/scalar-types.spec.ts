import axios from 'axios';

describe('GET /graphql', () => {
  it('should return a random ID', async () => {
    const query = `{ genRandomId }`;

    const res = await axios.post(
      '/graphql',
      { query },
      {
        headers: {
          'content-type': 'application/json',
        },
      },
    );

    expect(res.status).toBe(200);
    expect(typeof res.data.data.genRandomId).toBe('string');
  });

  it('should return a random mnemonic', async () => {
    const query = `{ genRandomMnemonic }`;

    const res = await axios.post(
      '/graphql',
      { query },
      {
        headers: {
          'content-type': 'application/json',
        },
      },
    );

    expect(res.status).toBe(200);
    expect(res.data.data.genRandomMnemonic.split(' ').length).toBe(
      12,
    );
    expect(typeof res.data.data.genRandomMnemonic).toBe('string');
  });

  it('should return a boolean', async () => {
    const query = `{ isItNightOnTheServer }`;

    const res = await axios.post(
      '/graphql',
      { query },
      {
        headers: {
          'content-type': 'application/json',
        },
      },
    );

    expect(res.status).toBe(200);
    expect(typeof res.data.data.isItNightOnTheServer).toBe('boolean');
  });

  it('should return the result of rolled dices', async () => {
    const query = `{ rollThreeDice }`;

    const res = await axios.post(
      '/graphql',
      { query },
      {
        headers: {
          'content-type': 'application/json',
        },
      },
    );

    expect(res.status).toBe(200);
    expect(res.data.data.rollThreeDice.length).toBe(3);
    for (const die of res.data.data.rollThreeDice) {
      expect(Number.isInteger(die)).toBeTruthy();
    }
  });
});
