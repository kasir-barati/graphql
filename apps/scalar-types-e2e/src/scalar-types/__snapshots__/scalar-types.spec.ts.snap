// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`POST /graphql should return partial data 1`] = `
{
  "data": {
    "something": {
      "getDie": null,
      "message": "whatever",
    },
  },
  "errors": [
    {
      "locations": [
        {
          "column": 13,
          "line": 7,
        },
      ],
      "message": "Unexpected error value: "FakeError"",
      "path": [
        "something",
        "getDie",
        "rollOnce",
      ],
    },
  ],
}
`;
